src/Parser/Serialization/DefaultSerializer.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+
+namespace Parser
+{
+
+    ///<summary>TODO combine with NLog's Json (injected) serializer?</summary>
```

Comment:
Yeah, I think it would be much better to use an existing JSON serializer:
- You have to write _tons_ of test cases to be sure you have a full, correct JSON implementation.
- Making it _fast_ is difficult and a lot of work.
- Then there's a ton of things you might want to support (e.g. look at Newtonsoft Json.NET): support for dictionaries, datatables, ignoring non-serializable members, etc. etc.
- And making it customizable is even worse. People invariably want lots of tweaks to naming, mappings, special types, omitting default values, etc. etc. Again just have a look at Json.NET, it's crazy.

Bottom line: I think logger and serializer should be two separated libs.
Even if you have ultra-basic internal JSON usage and don't need any fuss at all I would suggest taking a dependency on something like SimpleJSON.NET (probably not the case here).